data_steward/cdr_cleaner/cleaning_rules/negative_ages.py

```diff
@@ -27,110 +32,261 @@
 }
 
 person = common.PERSON
+death = common.DEATH
 MAX_AGE = 150
+affected_tables = list(date_fields.keys())
+affected_tables.append(death)
+
+#sandbox rows to be removed
+SANDBOX_NEGATIVE_AND_MAX_AGE_QUERY = JINJA_ENV.from_string("""
+CREATE OR REPLACE `{{project_id}}.{{sandbox_id}}.{{intermediary_table}}` AS (
+SELECT
+  *
+FROM
+  `{{project_id}}.{{dataset_id}}.{{table}}`
+WHERE
+  {{table}}_id IN (
+  SELECT
+    t.{{table}}_id
+  FROM
+    `{{project_id}}.{{dataset_id}}.{{table}}` t
+  JOIN
+    `{{project_id}}.{{dataset_id}}.{{person_table}}` p
+  ON
+    t.person_id = p.person_id
+  WHERE
+    t.{{table_date}} < DATE(p.birth_datetime)
+  UNION ALL
+  SELECT
+  t.{{table}}_id
+  FROM
+    `{{project_id}}.{{dataset_id}}.{{table}}` t
+  JOIN
+    `{{project_id}}.{{dataset_id}}.{{person_table}}` p
+  ON
+    t.person_id = p.person_id
+  WHERE
+    EXTRACT(YEAR
+    FROM
+      t.{{table_date}}) - EXTRACT(YEAR
+    FROM
+      p.birth_datetime) > {{MAX_AGE}})
```

Comment:
nit: `UNION ALL` can result in rows being duplicated in the sandbox table. Could we prevent this by using `UNION DISTINCT` or by ORing the criteria in a single statement like this?

```suggestion
SELECT
  *
FROM
  `{{project_id}}.{{dataset_id}}.{{table}}`
WHERE
  {{table}}_id IN (
  SELECT
    t.{{table}}_id
  FROM
    `{{project_id}}.{{dataset_id}}.{{table}}` t
  JOIN
    `{{project_id}}.{{dataset_id}}.{{person_table}}` p
  ON
    t.person_id = p.person_id
  WHERE
    t.{{table_date}} < DATE(p.birth_datetime)
    OR
    EXTRACT(YEAR
    FROM
      t.{{table_date}}) - EXTRACT(YEAR
    FROM
      p.birth_datetime) > {{MAX_AGE}})
```

Reply:
I've updated this to `UNION DISTINCT` please take a look and let me know if that is correct.